test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -88,3 +94,186 @@ def test_invalid_script_out(self):
+
+        # Check for the expected log message for insufficient resources
+        job_comment = "Can Never Run: Insufficient amount of server resource:"
+        job_comment += " foobar (R: 1 A: -1 T: -1)"
+
+        # The job shouldn't run
+        a = {'job_state': 'Q', 'comment': job_comment}
+        self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
+
+    def test_res_whitespace(self):
+        """
+        Test for parse errors when more than one white space
+        is added between the resource name and the !<script> in a
+        server_dyn_res line. There shouldn't be any errors.
+        """
+        # Create a resource of type long
+        resname = ["foo"]
+        restype = "long"
+
+        # Prep for server_dyn_resource scripts. Script "PbsPtl_get_foo*"
+        # generates file "PbsPtl_got_foo" and returns 1.
+        script_body = "echo get_foo > /tmp/PbsPtl_got_foo; echo 1"
+
+        fpath_out = os.path.join(os.sep, "tmp", "PbsPtl_got_foo")
+
+        (fd_in, fn_in) = self.du.mkstemp(prefix="PtlPbs_get_foo",
+                                         body=script_body)
+        self.du.chmod(path=fn_in, mode=0755, sudo=True)
+        os.close(fd_in)
+
+        # Add additional white space between resource name and the script
+        resval = ['"' + resname[0] + ' !' + fn_in + '"']
+
+        self.setup_dyn_res(resname, restype, resval)
+
+        # Check if the file "PbsPtl_got_foo" was created
+        time.sleep(15)
+        self.assertTrue(self.du.isfile(path=fpath_out, sudo=True))
+
+        # Submit job
+        a = {'Resource_List.foo': '1'}
+        jid = self.submit_job(a)
+
+        # Job must run successfully
+        a = {'job_state': 'R', 'Resource_List.foo': 1}
+        self.server.expect(JOB, a, id=jid)
+
+    def test_multiple_res(self):
+        """
+        Test multiple dynamic resources specified in resourcedef
+        and sched_config
+        """
+        # Create resources of type long
+        resname = ["foobar_small", "foobar_medium", "foobar_large"]
+        restype = "long"
+
+        # Prep for server_dyn_resource scripts.
+        script_body_s = "echo 8"
+        script_body_m = "echo 12"
+        script_body_l = "echo 20"
+
+        (fd_s, fn_s) = self.du.mkstemp(prefix="PtlPbs_small", suffix=".scr",
+                                       body=script_body_s)
+        (fd_m, fn_m) = self.du.mkstemp(prefix="PtlPbs_medium", suffix=".scr",
+                                       body=script_body_m)
+        (fd_l, fn_l) = self.du.mkstemp(prefix="PtlPbs_large", suffix=".scr",
+                                       body=script_body_l)
+
+        self.du.chmod(path=fn_s, mode=0755, sudo=True)
+        self.du.chmod(path=fn_m, mode=0755, sudo=True)
+        self.du.chmod(path=fn_l, mode=0755, sudo=True)
+
+        # Close file handles, else scheduler cannot execute them
+        os.close(fd_s)
+        os.close(fd_m)
+        os.close(fd_l)
+
+        resval = ['"foobar_small !' + fn_s + '"',
```

Comment:
use resname instead of foobar_small and rest

Reply:
Done